lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -5433,6 +5434,11 @@ void ByteCodeGenerator::EmitPropLoad(Js::RegSlot lhsLocation, Symbol *sym, Ident
             case Js::PropertyIds::undefined:
                 opcode = Js::OpCode::LdUndef;
                 break;
+            case Js::PropertyIds::__chakraLibrary:
+                if (CONFIG_FLAG(LdChakraLib)) {
+                    opcode = Js::OpCode::LdChakraLib;
```

Comment:
That's what is happening actually. When we generate the bytecode for the JS built ins function, the __chakraLibrary is loaded and gives access to ToLength, ToNumber...etc. Then, when we run ChakraCore without the flag -LdChakraLib, we don't load __chakraLibrary  anymore. But the JS Built ins methods still have access to it, privately.

---
In reply to: [115379529](https://github.com/Microsoft/ChakraCore/pull/2898#discussion_r115379529) [](ancestors = 115379529)